pkgs/development/tools/build-managers/gradle/default.nix

```diff
@@ -87,16 +102,13 @@ rec {
 
       fixupPhase = let arch = if stdenv.is64bit then "amd64" else "i386";
       in ''
+        . ${./patching.sh}
+
+        nativeVersion="$(extractVersion native-platform $out/lib/gradle/lib/native-platform-*.jar)"
         for variant in "" "-ncurses5" "-ncurses6"; do
-          mkdir "patching$variant"
-          pushd "patching$variant"
-          jar xf $out/lib/gradle/lib/native-platform-linux-${arch}$variant-${nativeVersion}.jar
-          patchelf \
-            --set-rpath "${stdenv.cc.cc.lib}/lib64:${lib.makeLibraryPath [ stdenv.cc.cc ncurses5 ncurses6 ]}" \
-            net/rubygrapefruit/platform/linux-${arch}$variant/libnative-platform*.so
-          jar cf native-platform-linux-${arch}$variant-${nativeVersion}.jar .
-          mv native-platform-linux-${arch}$variant-${nativeVersion}.jar $out/lib/gradle/lib/
-          popd
+          autoPatchelfInJar \
+            $out/lib/gradle/lib/native-platform-linux-${arch}$variant-''${nativeVersion}.jar \
```

Comment:
Just for the record - this is "just" patching the linux version (as it was done before). 
As others told me, the resulting nix package works on macOS too

Reply:
Yes, patchelf is a Linux-only thing, AFAIK.